src/sshagent/SSHAgent.cpp

```diff
@@ -402,13 +402,11 @@ void SSHAgent::removeAllIdentities()
 {
     auto it = m_addedKeys.begin();
     while (it != m_addedKeys.end()) {
-        // Remove key if requested to remove on lock
-        if (it.value().second) {
```

Comment:
If you remove this condition we will always remove _all_ keys on exit even with the bit not set which would be incorrect behavior.